docs/extensions.rst

```diff
@@ -0,0 +1,60 @@
+.. _extensions:
+
+##########
+Extensions
+##########
+
+Specialized extensions are provided for specific use cases.
+
+
+*****************
```

Comment:
Since supporting dataclasses and attrs is a stated goal, I'd rather break this into sub-pages rather than doing one huge page from the get go. Less likely to break links in the future.